scripts/rollover/cross-currency-rollover.ts

```diff
@@ -0,0 +1,275 @@
+    const nakamigos = <ERC721>erc721Factory.attach(COLLATERAL_ADDRESS);
+
+    const approveCollateralTx = await nakamigos.connect(borrower).approve(originationController.address, COLLATERAL_ID);
+    await approveCollateralTx.wait();
+
+    await originationController
+        .connect(borrower)
+        .initializeLoan(originalLoanTerms, borrowerStruct, originalLender.address, originalSig, sigProperties, []);
+
+    console.log(
+        `User ${borrower.address} borrowed ${PRINCIPAL} DAI at 15% interest from ${originalLender.address} against Nakamigos ${COLLATERAL_ID}`,
+    );
+
+    console.log(SECTION_SEPARATOR);
+
+    // advance time by two weeks into loan
+    await advanceTime(1209600); // 60 * 60 * 24 * 14
+
+    ////////////////////// MIGRATE LOAN TO NEW CURRENCY //////////////////////////////////////////////
+
+    console.log("Migrate loan from DAI currency to WETH currency ...");
+
+    // Calculate amount owed to original lender in the new currency:
+    // (PRINCIPAL + Interest rate ) / swap rate.
+    const interestAmount = await crossCurrencyRollover.calculateProratedInterestAmount(1);
+
+    const amountOwed = PRINCIPAL.add(interestAmount);
+    console.log("Total Amount Owed = Principal + Interest: ", ethers.utils.formatUnits(amountOwed, 18));
+
+    // price of Dai in wETH
+    const price = await crossCurrencyRollover.fetchCurrentPrice(DAIAddress, WETHAddress, 3000);
+    console.log("Price of Dai in wETH: ", ethers.utils.formatUnits(price, 18));
+
+    // amount owed in wETH
+    let NEW_PRINCIPAL = amountOwed.mul(price).div(ethers.utils.parseUnits("1", 18));
+
+    // account for 3% slippage
+    const slippage = NEW_PRINCIPAL.mul(3).div(100);
+    // NEW_PRINCIPAL + slippage amount for swap
+    NEW_PRINCIPAL = NEW_PRINCIPAL.add(slippage);
+    console.log("Amount Owed in New Currency Including Slippage: ", ethers.utils.formatUnits(NEW_PRINCIPAL, 18));
+
+    const newLoanTerms: LoanTerms = {
+        interestRate: INTEREST_RATE,
+        durationSecs: DURATION_SECS,
+        collateralAddress: COLLATERAL_ADDRESS,
+        deadline: Math.floor(Date.now() / 1000) + 60 * 60 * 24 * 7,
+        payableCurrency: NEW_PAYABLE_CURRENCY,
+        principal: NEW_PRINCIPAL,
+        collateralId: COLLATERAL_ID,
+        affiliateCode: AFFILIATE_CODE,
+    };
+
+    const newLenderSig = await createLoanTermsSignature(
+        crossCurrencyRollover.address,
+        "OriginationController",
+        newLoanTerms,
+        newLender,
+        EIP712_VERSION,
+        sigProperties,
+        "l",
+    );
+
+    console.log("Approvals for new loan...");
+    // new lender approves WETH amount to contract
+    const approveWETHTx = await weth.connect(newLender).approve(crossCurrencyRollover.address, wethAmount);
+    await approveWETHTx.wait();
+
+    // borrower approves borrower note
+    await borrowerNote.connect(borrower).approve(crossCurrencyRollover.address, 1);
+
+    await crossCurrencyRollover
```

Comment:
The script worked, which is great, but one request: can you print the starting and ending WETH and DAI balances for the borrower and lender before and after the rollover?